form/data_transfer_objects.rst

```diff
@@ -0,0 +1,316 @@
+.. index::
+   single: Form; Data Transfer Objects
+
+How to use Data Transfer Objects (DTOs)
+=======================================
+
+Data Transfer Objects can be used by forms to separate entities from the
+validation logic of forms.
+Entities should always have a valid state.
+When entities are used as data classes for a form, the data is injected into
+the entity and validated.
```

Comment:
When entities are used as data classes for a form, validation happens after the form data has been injected into the entities.